R/data_interface.R

```diff
@@ -239,10 +239,12 @@ spark_read_parquet <- function(sc,
                                columns = NULL,
                                schema = NULL,
                                ...) {
-  c(name, path) %<-% spark_read_compat_param(sc, name, path)
+  params <- spark_read_compat_param(sc, name, path)
+  name <- params[1L]
+  path <- params[-1L]
```

Comment:
Did you hit a particular issue with:

```r
c(name, path) %<-% spark_read_compat_param(sc, name, path)
```

Otherwise, can we revert this and other changes to `c(name, path) %<-%`?

Reply:
Yes, there is a check in `%<-%` that would fail here as it would expect length 2 on both sides:
```
if (length(dest) != length(value)) stop("Assignment must contain same number of elements")
```

Interestingly enough, the [zeallot package](https://github.com/r-lib/zeallot) implements the same operator and has a construct for similar cases:
```
c(first, ...rest) %<-% list("a", "b", "c", "d", "e")
```
It also is a (recursive) dependency of sparklyr, but I did not want to make such a strong change here, so I went with the base solution above.